src/Microsoft.AspNetCore.Server.Kestrel/Http/DateHeaderValueManager.cs

```diff
@@ -55,19 +55,19 @@ public DateHeaderValueManager()
         /// Returns a value representing the current server date/time for use in the HTTP "Date" response header
         /// in accordance with http://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.18
         /// </summary>
-        /// <returns>The value.</returns>
-        public virtual string GetDateHeaderValue()
+        /// <returns>The value in string and byte[] format.</returns>
+        public DateHeaderValues GetDateHeaderValues()
         {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _dateValue;
-        }
+            if (!_hadRequestsSinceLastTimerTick)
+            {
+                PrepareDateValues();
+            }
 
-        public byte[] GetDateHeaderValueBytes()
-        {
-            _hadRequestsSinceLastTimerTick = true;
-            PrepareDateValues();
-            return _activeDateBytes ? _dateBytes0 : _dateBytes1;
+            return new DateHeaderValues()
+            {
+                Bytes = _activeDateBytes ? _dateBytes0 : _dateBytes1,
```

Comment:
Assuming it's fine, you might be able to eliminate a read and a branch here by having the updater set an _activeBytes field to either _dateBytes0 or _dateBytes1, such that the reader can then just do ```Bytes = _activeBytes```.  Assuming reading is super hot and updating is rare, avoiding branches and extra reads on the super hot path could be beneficial.